backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -1092,4 +1105,188 @@ export class FeatureFlagService {
+      const isCompatible = validation && validation.compatibilityType !== FF_COMPATIBILITY_TYPE.INCOMPATIBLE;
+
+      return {
+        fileName: file.fileName,
+        error: isCompatible ? validation.compatibilityType : FF_COMPATIBILITY_TYPE.INCOMPATIBLE,
+      };
+    });
+
+    const validFiles: ImportFeatureFlagListValidator[] = fileStatusArray
+      .filter((fileStatus) => fileStatus.error !== FF_COMPATIBILITY_TYPE.INCOMPATIBLE)
+      .map((fileStatus) => {
+        const featureFlagListFile = featureFlagListFiles.find((file) => file.fileName === fileStatus.fileName);
+        return JSON.parse(featureFlagListFile.fileContent as string);
+      });
+
+    const createdLists: (FeatureFlagSegmentInclusion | FeatureFlagSegmentExclusion)[] =
+      await this.dataSource.transaction(async (transactionalEntityManager) => {
+        const listDocs: FeatureFlagListValidator[] = [];
+        for (const list of validFiles) {
+          const listDoc: FeatureFlagListValidator = {
+            ...list,
+            enabled: false,
+            flagId: featureFlagId,
+            segment: { ...list.segment, id: uuid() },
+          };
+
+          listDocs.push(listDoc);
+        }
+
+        if (listType === FEATURE_FLAG_PARTICIPANT_LIST_KEY.INCLUDE) {
+          return await this.addList(
+            listDocs,
+            FEATURE_FLAG_LIST_FILTER_MODE.INCLUSION,
+            currentUser,
+            logger,
+            transactionalEntityManager
+          );
+        } else {
+          return await this.addList(
+            listDocs,
+            FEATURE_FLAG_LIST_FILTER_MODE.EXCLUSION,
+            currentUser,
+            logger,
+            transactionalEntityManager
+          );
+        }
+      });
+
+    logger.info({ message: 'Imported feature flags', details: createdLists });
+
+    fileStatusArray.forEach((fileStatus) => {
+      if (fileStatus.error !== FF_COMPATIBILITY_TYPE.INCOMPATIBLE) {
+        fileStatus.error = null;
+      }
+    });
+    return fileStatusArray;
+  }
+
+  public async validateImportFeatureFlagLists(
+    featureFlagFiles: IFeatureFlagFile[],
+    featureFlagId: string,
+    logger: UpgradeLogger
+  ): Promise<ValidatedFeatureFlagsError[]> {
+    logger.info({ message: 'Validate feature flag lists' });
+
+    const parsedFeatureFlagLists = featureFlagFiles.map((featureFlagFile) => {
+      try {
+        return {
+          fileName: featureFlagFile.fileName,
+          content: JSON.parse(featureFlagFile.fileContent as string),
+        };
+      } catch (parseError) {
+        logger.error({ message: 'Error in parsing feature flag file', details: parseError });
+        return {
+          fileName: featureFlagFile.fileName,
+          content: null,
+        };
+      }
+    });
+
+    const featureFlag = await this.findOne(featureFlagId, logger);
+
+    const validationErrors = await Promise.allSettled(
+      parsedFeatureFlagLists.map(async (parsedFile) => {
+        if (!featureFlag) {
```

Comment:
Write a single if statement 
```suggestion
        if (!featureFlag || (!parsedFile.content))
```